pyomo/gdp/basic_step.py

```diff
@@ -26,10 +30,23 @@ def _pseudo_clone(self):
     return new_block
 
 
-def apply_basic_step( disjunctions ):
+def _squish_singletons(tuple_iter):
+    """Squish all singleton tuples into their non-tuple values."""
+    for tup in tuple_iter:
+        if len(tup) == 1:
+            yield tup[0]
+        else:
+            yield tup
```

Comment:
Why not use `flatten_tuple`?

Reply:
Actually, looking at what `flatten_tuple` does, we only want to flatten it if it's a singleton. That is, `((0, 0), (0, 1), (0, 2))` is fine, but when only one disjunction is participating, it becomes `((0,), (1,), (2,))`, which is problematic as input to `initialize=` for `Set(dimen=1)`.